search/views.py

```diff
@@ -105,13 +107,35 @@ def syllabus(request, id):
                   {'course': course, 'syllabus_path': request.path})
 
 
-def hit(request, id):
-    course = get_object_or_404(Course, id=id)
-    course.hit += 1
-    course.save()
+def course_manipulation(request, id):
+    """ Use this `course_manipulation` function to record the course """
+    if request.user.is_authenticated():
+        course = get_object_or_404(Course, id=id)
+        request_type = request.GET.get('type', 'GET')
+
+        if request_type == 'PUT':
+            request.user.member.courses.add(course)
+        elif request_type == 'DELETE':
+            request.user.member.courses.remove(course)
```

Comment:
django supports ``request.method == 'PUT'`` but not ``request.PUT``

actually, if you do this
```bash
curl server_address -X any_method_name
```
you will get
```python
request.method == 'ANY_METHOD_NAME'
```
in django